datafusion/physical-expr/src/aggregate/variance.rs

```diff
@@ -286,17 +286,17 @@ impl Accumulator for VarianceAccumulator {
             }
         };
 
-        if count <= 1 {
-            return Err(DataFusionError::Internal(
-                "At least two values are needed to calculate variance".to_string(),
-            ));
-        }
-
-        if self.count == 0 {
-            Ok(ScalarValue::Float64(None))
-        } else {
-            Ok(ScalarValue::Float64(Some(self.m2 / count as f64)))
-        }
+        Ok(ScalarValue::Float64(match self.count {
+            0 => None,
+            1 => {
+                if matches!(self.stats_type, StatsType::Population) {
```

Comment:
Nit: In this case I think `if let` would be slightly more idiomatic.

Reply:
done in [bc65eeb](https://github.com/apache/arrow-datafusion/pull/4847/commits/bc65eeb0300b6e330b0a9963d961ecea5bd760c2)